cli/checker-cli-parser/src/main/scala/com/rackspace/com/papi/components/checker/cli/CheckerParser.scala

```diff
@@ -0,0 +1,172 @@
+package com.rackspace.com.papi.components.checker.cli
+
+import com.rackspace.com.papi.components.checker.Config
+import scopt.OptionParser
+
+import scala.language.implicitConversions
+
+class CheckerParser(programName: String) extends OptionParser[Config](programName) {
+
+  opt[Unit]('d', "remove-dups").text("Remove duplicate nodes. Default: false")
+    .action((_, c) => {
+      c.removeDups = true; c
+    })
```

Comment:
Since we are mutating the config object itself, I still like simplifying this by using the mutable parsing approach and just passing the configuration object in as a constructor argument. I'm not sure the immutable approach buys us much, but it does force us to do this thing where we mutate then return the mutated object everywhere.

So I propose changing the few lines above to:
```
class CheckerParser(programName: String, c: Config) extends OptionParser[Unit](programName) {

  opt[Unit]('d', "remove-dups").text("Remove duplicate nodes. Default: false")
    .foreach(_ => c.removeDups = true)
```

Reply:
@wdschei Any comment on this?

Reply:
I spent about an hour looking at the Scopt examples and the way this is done in the `Wadl2Checker` class.
Yes, way longer that was probably necessary, but I kept going back and forth.
I ended up coming to the conclusion that the `.foreach` is a little shorter, but the `.action` makes it easier to follow.
This is probably due to the `.action` being the Scopt paradigm and the docs lending themselves to it.
In the end it was a total tossup and I didn't feel strongly either way.

Reply:
Now as I'm thinking about it, adding a new `apply` method that pseudo-mimics the `copy` method would work too.
Instead of `copy`ing the object and returning it, just `apply` the changes and return it like a `builder`.

Reply:
To be fair, Scopt has multiple operating paradigms, so I'm not sure it's fair to call `.action` **the** paradigm. The `.foreach` paradigm has a short documentation segment, but since it is essentially just a replacement for `.action`, I personally find it sufficient (https://github.com/scopt/scopt#object-oriented-dsl-mutable-parsing). Plus, the `.action` paradigm really seems like its intended to operator on `case class`, which is not what we have in `Config`.

My thinking was that if we have a mutable object, mutate it. It lets us continue working with the `Config` class without the need for modification in a way that is still documented and supported by Scopt without the unnecessary verbosity of using `.action`.

As for an `apply` method, I think that route would effectively have you writing the same code that is generated by using `case class` instead of `class`. That is, at that point, I would suggest just converting `Config` into a `case class` and then using `.action` and the immutable paradigm.

I think either of those two approaches would be clearer and more concise than what we have here (which mixes the immutable parsing paradigm with a mutable `Config` object).

Reply:
Fair enough.
I guess where I'm coming from is that the mutable parsing way is obviously not the preferred Scopt way, or it wouldn't be the tiniest blurb at the bottom of the Readme.
That said, if we are going to use the non-preferred mutable way, then we should probably use their prescribed `.foreach` way of doing it.
I'm not supper hard over on this, but for consistency we should probably use only one way.